internal/buffer/buffer.go

```diff
@@ -150,9 +150,11 @@ func (b *SharedBuffer) MarkModified(start, end int) {
 	end = util.Clamp(end, 0, len(b.lines)-1)
 
 	if b.Settings["syntax"].(bool) && b.SyntaxDef != nil {
+		l := start
 		for i := start; i <= end; i++ {
```

Comment:
Side note: it seems we have redundant steps in this loop, which we can avoid by changing `i++` to `i = l + 1`.